pkg/resources/grant_privileges_to_database_role.go

```diff
@@ -0,0 +1,1031 @@
+package resources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// TODO: Handle IMPORTED PRIVILEGES privilege (after second account will be added - SNOW-976501)
+
+var grantPrivilegesToDatabaseRoleSchema = map[string]*schema.Schema{
+	"database_role_name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		Description:      "The fully qualified name of the database role to which privileges will be granted.",
+		ValidateDiagFunc: IsValidIdentifier[sdk.DatabaseObjectIdentifier](),
+	},
+	"privileges": {
+		Type:        schema.TypeSet,
+		Optional:    true,
+		Description: "The privileges to grant on the database role.",
+		ExactlyOneOf: []string{
+			"privileges",
+			"all_privileges",
+		},
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			ValidateDiagFunc: isNotOwnershipGrant(),
+		},
+	},
+	"all_privileges": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Grant all privileges on the database role.",
+		ExactlyOneOf: []string{
+			"privileges",
+			"all_privileges",
+		},
+	},
+	"with_grant_option": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		ForceNew:    true,
+		Description: "If specified, allows the recipient role to grant the privileges to other roles.",
+	},
+	"always_apply": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "If true, the resource will always produce a “plan” and on “apply” it will re-grant defined privileges. It is supposed to be used only in “grant privileges on all X’s in database / schema Y” or “grant all privileges to X” scenarios to make sure that every new object in a given database / schema is granted by the account role and every new privilege is granted to the database role. Important note: this flag is not compliant with the Terraform assumptions of the config being eventually convergent (producing an empty plan).",
+	},
+	"always_apply_trigger": {
+		Type:        schema.TypeString,
```

Comment:
At first, it was a boolean, but I thought it could be acting wrongly in some scenarios. For example, the only place where this flag could be switched is in the Read operation (because it is invoked on every tf action), but then we could end up in a situation where sometimes it triggers and sometimes doesn't (mostly because Update operation calls Read at the end). With random strings, we don't have to think about it and wonder if it will be switched correctly because it will be always a different value.